async/deferred.ts

```diff
@@ -5,6 +5,7 @@
 // See https://github.com/Microsoft/TypeScript/issues/15202
 // At the time of writing, the github issue is closed but the problem remains.
 export interface Deferred<T> extends Promise<T> {
+  status: "pending" | "fulfilled" | "rejected";
```

Comment:
In the ECMAScript specification, the internal field is called `[[PromiseState]]`, so I think `state` would be better name.